analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ParametersCorrectOrder.cs

```diff
@@ -193,4 +193,65 @@ public void M3()
             double doTheThing(int divisor, int dividend) => divide(dividend, divisor);  // Noncompliant
         }
     }
+
+    // See https://github.com/SonarSource/sonar-dotnet/issues/3879
+    class NotOnlyNullableParam
+    {
+        internal class A
+        {
+            public C Something { get; set; }
+        }
```

Comment:
As the current implementation is not sensitive to `int?` type, can you add `Value` property here and use it in compliant and noncompliant way?

Reply:
Same for VB please

Reply:
Suddenly I realized that I could not distinguish between Nullable <T> .Value and SomeClass.Value (property) in expressions.
In the first case, you need to take the internal expression, and in the second, it is itself.
Can you suggest how this can be done?

Reply:
I'm sorry for coming back to this late. I've lost the context a little meanwhile. Can you push a commit with (failing) UT that will demonstrate your intention? We can come up with a solution from that.